### DISCUSSION TITLE
SPA mode does not work in a production environment with forced HTTPS?

### DISCUSSION THREAD

**Discussion by robinheidrich** [filamentphp/filament#9979] [Help]
It works locally, but not on the production server with HTTPS enforced. If HTTPS is not enforced, it will work on the production server.

I could not find any issue or discussion about this.

Has anyone had the same problem and found a solution?

**Comment by leandrocfe** [comment]
What is the error?

**Reply by robinheidrich** [reply]
There is no error, just no `wire:navigate` set for the links.

**Reply by suleymanozev** [reply]
I think there is a problem here too

**Comment by Solunsky** [comment]
Read: [StackOverflow](https://stackoverflow.com/a/70642976)
See: function **is_app_url()** in _vendor/filament/support/src/helpers.php_

Not correct solution, but work, add to provider:
```
$this->app['request']->server->set('HTTPS', 'on');
URL::forceScheme('https');
```

**Reply by liquid207** [reply]
@SalvaTerol Thanks, save my day

**Reply by kucingtakucin** [reply]
still not working for me

**Reply by denizaygundev** [reply]
I think I realised what's causing this, is everyone here using Cloudflare or a similar service in a "flexible" mode? 

"Enable encryption only between your visitors and Cloudflare. This will avoid browser security warnings, but all connections between Cloudflare and your origin are made through HTTP."

**Reply by denizaygundev** [reply]
@kucingtakucin @robinheidrich 

The proper solution for me, wasn't to hack in any forced https schema's but to ensure:
1. SSL certificate setup in Forge (in my case because I need multiple nested subdomains `https://hello.api.example.com`, I obtained a Let's Encrypt against `api.example.com` and `*.api.example.com` using Cloudflare as the DNS provider whilst paying for Total TLS so Let's Encrypt can order an advanced certificate against multi-level sub-domains. 
2. Ensure Cloudflare was configured properly with some HTTP to HTTPS redirect features enabled

The Forge docs also helped: https://forge.laravel.com/docs/sites/ssl
